File: 6_PYTHON/cash/cash.py
from cs50 import get_float

coins = [25, 10, 5, 1]


def main():
    change = int(get_valid_float() * 100)
    change_coins = 0
    for coin in coins:
        if change >= coin:
            num_coins = change // coin
            change -= num_coins * coin
            change_coins += num_coins
    print(change_coins)


def get_valid_float():
    while True:
        change = get_float("Change owed: ")
        if change > 0:
            break
    return change


main()

File: cash/cash.c
#include <cs50.h>
#include <stdio.h>

float get_valid_change(void);
int calcule_change_coin_amount(float change);

int main(void)
{
    float change = get_valid_change();
    int coinAmount = calcule_change_coin_amount(change);
    printf("%i\n", coinAmount);
}

int calcule_change_coin_amount(float change)
{
    int coins[4] = {25, 10, 5, 1};
    int coinAmount = 0;
    int convertedChange = (int) (change * 100 + 0.5);
    while (convertedChange > 0)
    {
        for (int i = 0; i < 4; i++)
        {
            if (convertedChange - coins[i] >= 0)
            {
                convertedChange -= coins[i];
                coinAmount++;
                i--;
            }
        }
    }
    return coinAmount;
}

float get_valid_change(void)
{
    float change;
    do
    {
        change = get_float("Troca devida: ");
    }
    while (change < 0);
    return change;
}
